model/database/cosmosdb-metrics.yaml

```diff
@@ -0,0 +1,57 @@
+groups:
+  - id: metric.db.cosmosdb.operation.request_charge
+    type: metric
+    metric_name: db.cosmosdb.operation.request_charge
+    brief: "Request charge consumed by the operation."
+    instrument: histogram
+    unit: "{request_units}"
+    stability: experimental
+    extends: attributes.db.cosmosdb.minimal
+
+  - id: metric.db.cosmosdb.operation.regions_contacted
+    type: metric
+    metric_name: db.cosmosdb.operation.regions_contacted
+    brief: "Number of regions contacted for the operation"
+    instrument: histogram
+    unit: "{count}"
+    stability: experimental
+    extends: attributes.db.cosmosdb.minimal
+
+  - id: metric.db.cosmosdb.client.active_instances
```

Comment:
do you really want to capture the number of clients? It seems to be a proxy to some other metric that actually matters like number of active connections.

I.e. I assume that if clients used the same connection or the same connection pool, then number of clients would not really matter in the context of CosmosDB (it would only be interesting in the context of garbage collection and avoiding extra allocations).

If tomorrow you change the design of the SDK and start sharing connections, chances are that number of active clients will become useless metric and you'd need to create another one.

Let me know if I'm missing something.

BTW https://github.com/open-telemetry/semantic-conventions/blob/093623eae80537697a89681132f9e18fa38257f9/docs/database/database-metrics.md#metric-dbclientconnectioncount already measures number of active/idle connections and maybe it would be useful to just report this metric?

Reply:
This metric does not represent 'active connections'; instead, it shows the 'number of SDK client instances.' A common issue occurs when customers, either knowingly or unknowingly, maintain multiple instances of the Cosmos SDK client. This leads to the creation of multiple caches, triggering frequent cache refresh calls, which can result in high CPU and memory consumption. Monitoring the number of active client instances is therefore crucial.

Reply:
+1, while we do have a connection sharing feature in mind for the future, number of active clients is still useful and having more than one client per account is an antipattern